### DISCUSSION TITLE
Local Repository

### DISCUSSION THREAD

**Discussion by pateldm2008** [microsoft/winget-cli#1554] [Q&A] [answered]
Can we host Package Repository internally? We like to use winget to install Infrastructure Software/Tools on Images or newly built Windows Servers.

**Comment by denelon** [comment, marked as answer]
Yes, we have a reference implementation at https://github.com/microsoft/winget-cli-restsource.

**Reply by denelon** [reply]
Note, Windows Server 2019 is not supported due to several dependencies the Windows Package Manager has. Also support on Windows Server 2022 should also be treated as "experimental" only. The license for offline installation is available on the releases page.

**Comment by mherrmann** [comment]
[winget.Pro](https://winget.pro) also gives you a loal repository for self-hosting.
